playwright-e2e/pages/SelectDateTimePage.ts

```diff
@@ -78,14 +76,51 @@ export default class SelectDateTimePage extends BasePage {
     }
 
     async selectFirstAvailableSlot(): Promise<void> {
-        await this.showAllSlots.first().click()
-        const firstAvailableSlot = this.availableSlot.first()
-        await firstAvailableSlot.check()
-        expect(await firstAvailableSlot.isChecked()).toBeTruthy()
+        // Find all day links in the calendar
+        const dayLinks = this.page.locator('a[id^="day-link-"]')
+
+        const count = await dayLinks.count()
+
+        for (let i = 0; i < count; i++) {
+            const link = dayLinks.nth(i)
+
+            // Click the calendar day link
+            await link.click()
+
+            // Extract the date string from the link ID
+            const linkId = await link.getAttribute('id')
+            const dayGroupId = linkId?.replace('day-link', 'day-group')
+
+            if (!dayGroupId) continue
+
+            // Locate radio buttons inside the revealed date group
+            const slotRadioButtons = this.page.locator(`#${dayGroupId} input[type="radio"]`)
+
+            // Check if at least one slot exists
+            if (await slotRadioButtons.count() > 0) {
+                const firstAvailableSlot = slotRadioButtons.first()
+                await firstAvailableSlot.scrollIntoViewIfNeeded()
+                await firstAvailableSlot.check()
+                expect(await firstAvailableSlot.isChecked()).toBeTruthy()
+
```

Comment:
This might fail to find an available slot if it is only checking for the first radio button. For example: morning and afternoon session on a day. If the prisoner already has a morning visit then that radio will be **disabled** but the afternoon one might be available.

You could get around this by looking for the first **non-disabled** radio.

Equally this might not be a problem if the test is set up so there is always a fair amount of capacity.